src/expression_evaluator/function_evaluator.cpp

```diff
@@ -25,7 +26,9 @@ void FunctionExpressionEvaluator::evaluate(ClientContext* clientContext) {
     }
     auto expr =
         ku_dynamic_cast<binder::Expression*, binder::ScalarFunctionExpression*>(expression.get());
-    if (expr->getFunctionName() == function::CastAnyFunction::name) {
+    if (expr->getFunctionName() == function::CastAnyFunction::name ||
```

Comment:
Open an issue to remove this if statement. We should have an execFunc that passes FunctionBindData only or make clientContext part of FunctionBindData

Reply:
I'll open an issue about it. Doing this will also allow us to use ```constPtrCast()``` to downcast ```dataPtr``` to ```FunctionStringBindData``` and ```CastFunctionBindData```.